api/src/main/java/javax/enterprise/inject/Instance.java

```diff
@@ -160,7 +161,31 @@
      * @throws IllegalArgumentException if passed two instances of the same qualifier type, or an instance of an annotation that
      *         is not a qualifier type
      */
-    public <U extends T> Instance<U> select(TypeLiteral<U> subtype, Annotation... qualifiers);
+    <U extends T> Instance<U> select(TypeLiteral<U> subtype, Annotation... qualifiers);
+
+    /**
+     * <p>
+     * When called, provides back a Stream of the beans available in this Instance.  If no beans are found, it
+     * returns an empty stream.
+     * </p>
+     *
+     * @return a <tt>Stream</tt> representing the beans associated with this {@link Instance} object
+     */
+    default Stream<T> stream(){
+        return StreamSupport.stream(this.spliterator(), false);
+    }
+
+    /**
+     * <p>
+     * When called, provides back a Stream of the beans available in this Instance.  If no beans are found, it
+     * returns an empty stream.
+     * </p>
+     *
+     * @return a <tt>Stream</tt> representing the beans associated with this {@link Instance} object
+     */
+    default Stream<T> parallelStream(){
```

Comment:
Not sure I am following here, how would you make it "container aware". 

Btw @tremes raised a good point - can we allow parallel processing at all? It spawns new threads in which we **do not** propagate context. Therefore if there are some, say, session scoped beans in the Instance, their context would not work in other thread. Or am I missing something?